lib/src/Uncertainty/StatTests/LinearModelTest.cxx

```diff
@@ -46,15 +50,69 @@ TestResult LinearModelTest::LinearModelFisher(const Sample & firstSample,
     const LinearModel & linearModel,
     const Scalar level)
 {
-  return RunTwoSamplesALinearModelRTest(firstSample, secondSample, linearModel, level, "LmFisher");
+  const UnsignedInteger dimension = firstSample.getDimension();
+  if (secondSample.getDimension() != 1) throw InvalidDimensionException(HERE) << "Error: output sample must be 1D";
+  if (firstSample.getSize() != secondSample.getSize()) throw InvalidArgumentException(HERE) << "Error: input and output samples must have the same size";
+  if (firstSample.getSize() != secondSample.getSize()) throw InvalidArgumentException(HERE) << "Error: input and output samples must have the same size";
+  if (firstSample.getSize() < 3) throw InvalidArgumentException(HERE) << "Error: sample too small. Sample should contains at least 3 elements";
+  const UnsignedInteger size = firstSample.getSize();
+
+  // Regression coefficient
+  LinearCombinationFunction fHat(LinearBasisFactory(dimension).build(), linearModel.getRegression());
+  const Sample yHat(fHat(firstSample)); 
+  const Sample residualSample(secondSample - yHat);
+  
+  // For the Fisher test, we need both Sum of Squared Explained (SSE)
+  // and the Sum of Squared Residuals
+
+  const Scalar sumSquaredExplained = (yHat - secondSample.computeMean()).computeRawMoment(2)[0] * firstSample.getSize() ;
+  const Scalar sumSquaredResiduals = residualSample.computeRawMoment(2)[0] * firstSample.getSize() ;
+
+
+  // The statistical test checks the nullity of the regression linear model coefficients
+  // H0 : Beta_i = 0
+  // H1 : Beta_i < 0 or Beta_i > 0
+  // The statistics follows a Fisher distribution
+  const Scalar statistic = sumSquaredResiduals / dimension / (sumSquaredExplained / (size - dimension - 1));
```

Comment:
Should check that size - dimension - 1>0 otherwise throw an InvalidArguentException.